src/switch_core_file.c

```diff
@@ -62,6 +62,78 @@ static switch_status_t get_file_size(switch_file_handle_t *fh, const char **stri
 	return status;
 }
 
+SWITCH_DECLARE(switch_status_t) switch_core_file_exists(const char *file_path, switch_memory_pool_t *pool)
+{
+	switch_status_t status;
+	char *rhs = NULL;
+	if (zstr(file_path)) {
+		return SWITCH_STATUS_FALSE;
+	}
+	if (! pool) {
+		if ((status = switch_core_new_memory_pool(&pool)) != SWITCH_STATUS_SUCCESS) {
+			return status;
+		}
+	}
```

Comment:
You never free this pool ! If you created it, you must delete it before the end of the function. So I would define new_pool, and pool = new_pool, and at the end if (new_pool) delete new_pool.  You will have to change it to use goto instead of return as well after these lines